ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/strategy/channelstrategy/SequentialChannelStrategy.java

```diff
@@ -109,10 +115,19 @@ public void releaseChannelForBlob(final SeekableByteChannel seekableByteChannel,
 
             blobNameOffsetMap.remove(blobName, blob.getOffset());
 
-            if (blobNameOffsetMap.get(blobName).size() == 0) {
+            final Long maximumOffset = masterObjectList.getObjects().stream()
+                    .flatMap(objects -> objects.getObjects().stream())
+                    .filter(bulkObject -> bulkObject.getName().equals(blobName))
+                    .map(bulkObject -> bulkObject.getOffset())
+                    .max(Long::compareTo).orElseGet(() -> blob.getOffset());
+
+            final boolean isReadOnly = ((SeekableByteChannelDecorator) seekableByteChannel).wrappedSeekableByteChannel() instanceof ReadOnlySeekableByteChannel;
```

Comment:
Why is the cast here needed in order to perform the `instanceof` check?

Reply:
We know it is a SeekableByteChannelDecorator because we wrap it further up, but the wrappedSeekableByteChannel() is only on the Decorator